### PR TITLE
Don't deploy `PSP`s when `PodSecurityPolicy` plugin is disabled 

### PR DESCRIPTION
**Pull request by shafeeqes**
**What this PR does / why we need it**:
`PodSecurityPolicy` is deprecated and will be disabled in kubernetes v1.25+.
End-users are provided an option to migrate their PSPs before upgrading and disable the `PodSecurityPolicy` admission plugin in the ShootSpec. 
In that case, we should stop deploying our PSPs as well.

- This PR stops deploying PSPs related to `shoot-networking-problemdetector` if the plugin is disabled in the Shoot.

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/5250

**Special notes for your reviewer**:

**Release note**:

```breaking operator
Please make sure you're running gardener@v1.52 or above before upgrading to this version.
```

### CODE REVIEW

File: pkg/controller/lifecycle/actuator.go [hunk 1]
```diff
@@ -106,20 +119,16 @@ func (a *actuator) createSeedResources(ctx context.Context, log logr.Logger, clu
...
 		if a.serviceConfig.NetworkProblemDetector.DefaultPeriod != nil {
 			defaultPeriod = a.serviceConfig.NetworkProblemDetector.DefaultPeriod.Duration
 		}
-		if a.serviceConfig.NetworkProblemDetector.PSPDisabled != nil {
-			pspEnabled = !*a.serviceConfig.NetworkProblemDetector.PSPDisabled
-		}
 	}
 
-	shootResources, err := a.getShootAgentResources(defaultPeriod, pingEnabled, pspEnabled)
+	shootResources, err := a.getShootAgentResources(defaultPeriod, pingEnabled, !pspDisabled)
```
Comment (MartinWeindel) [inline pkg/controller/lifecycle/actuator.go:131]: Please restore the former behaviour if `pspDisabled == false`. In this case it should be possible to disable PSP by the service config, i.e. supposed change:

```go
	pspDisabledByConfig := false
	pingEnabled := false
	if a.serviceConfig.NetworkProblemDetector != nil {
		if a.serviceConfig.NetworkProblemDetector.DefaultPeriod != nil {
			defaultPeriod = a.serviceConfig.NetworkProblemDetector.DefaultPeriod.Duration
		}
		if a.serviceConfig.NetworkProblemDetector.PSPDisabled != nil {
			pspDisabledByConfig = *a.serviceConfig.NetworkProblemDetector.PSPDisabled
		}
	}

	shootResources, err := a.getShootAgentResources(defaultPeriod, pingEnabled, !pspDisabled && !pspDisabledByConfig)
```

**Review by MartinWeindel — CHANGES_REQUESTED** [review]
The configuration parameter serviceConfig.NetworkProblemDetector.PSPDisabled should still be used, if PSP is not disabled by the shoot.

File: pkg/controller/lifecycle/actuator.go [hunk 2]
```diff
@@ -66,8 +68,19 @@ func (a *actuator) Reconcile(ctx context.Context, log logr.Logger, ex *extension
 		return err
 	}
 
+	pspDisabled := false
+	if a.serviceConfig.NetworkProblemDetector != nil {
+		if a.serviceConfig.NetworkProblemDetector.PSPDisabled != nil {
+			pspDisabled = *a.serviceConfig.NetworkProblemDetector.PSPDisabled
+		}
+	}
+
```
Comment (shafeeqes) [inline pkg/controller/lifecycle/actuator.go:77]: The former behaviour still stands. I just moved it out.
But if it's cleaner your way, then let's go for it.

File: pkg/controller/lifecycle/actuator.go (hunk 1 above)
Comment (shafeeqes) [reply pkg/controller/lifecycle/actuator.go]: The former behaviour still stands. I just moved it out.
But if it's cleaner your way, then let's go for it.

**Review by MartinWeindel — APPROVED** [review]
/lgtm

**Pull request merged**